lib/libc/common/source/time/asctime_r.c

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright (c) 2024 Meta Platforms
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <stdio.h>
+#include <string.h>
+#include <time.h>
+
+#include <zephyr/sys/util.h>
+
+#define DATE_STRING_BUF_SZ 26
+
+char *asctime_r(const struct tm *tp, char *buf)
+{
+	static const char wday_str[7][3] = {
+		"Sun", "Mon", "Tue", "Wed", "Thu", "Fri", "Sat",
+	};
+	static const char mon_str[12][3] = {
+		"Jan", "Feb", "Mar", "Apr", "May", "Jun", "Jul", "Aug", "Sep", "Oct", "Nov", "Dec",
+	};
+
+	if ((tp == NULL) || (buf == NULL) || (tp->tm_wday >= ARRAY_SIZE(wday_str)) ||
+	    (tp->tm_mon >= ARRAY_SIZE(mon_str))) {
+		return NULL;
+	}
+
+	snprintf(buf, DATE_STRING_BUF_SZ, "%.3s %.3s%3d %.2d:%.2d:%.2d %d\n", wday_str[tp->tm_wday],
```

Comment:
You could easily add a check here to see if there is a buffer overflow and return NULL in that case; that's what glibc and picolibc (just recently) do. (valid returns from snprintf must be strictly less than DATE_STRING_BUF_SZ)